test/k8sT/Services.go

```diff
@@ -112,83 +112,111 @@ var _ = Describe("K8sValidatedServicesTest", func() {
-		demoDSPath := kubectl.ManifestGet("demo.yaml")
-		kubectl.Apply(demoDSPath)
-		defer kubectl.Delete(demoDSPath)
+	Context("Checks ClusterIP Connectivity", func() {
 
-		pods, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
-		Expect(pods).Should(BeTrue())
-		Expect(err).Should(BeNil())
+		var (
+			demoYAML = kubectl.ManifestGet("demo.yaml")
+		)
 
-		svcIP, err := kubectl.Get(
-			helpers.DefaultNamespace, fmt.Sprintf("service %s", serviceName)).Filter("{.spec.clusterIP}")
-		Expect(err).Should(BeNil())
-		Expect(govalidator.IsIP(svcIP.String())).Should(BeTrue())
+		BeforeEach(func() {
+			res := kubectl.Apply(demoYAML)
+			res.ExpectSuccess("unable to apply %s: %s", demoYAML, res.CombineOutput())
+		})
 
-		status := kubectl.Exec(fmt.Sprintf("curl http://%s/", svcIP))
-		status.ExpectSuccess()
+		AfterEach(func() {
+			// Explicitly ignore result of deletion of resources to avoid incomplete
+			// teardown if any step fails.
+			_ = kubectl.Delete(demoYAML)
+		})
 
-		ciliumPod, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
-		Expect(err).Should(BeNil())
+		It("Checks service on same node", func() {
+			pods, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+			Expect(pods).Should(BeTrue())
+			Expect(err).Should(BeNil())
 
-		service := kubectl.CiliumExec(ciliumPod, "cilium service list")
-		Expect(service.Output()).Should(ContainSubstring(svcIP.String()))
-		service.ExpectSuccess()
+			res, err := kubectl.Get(
+				helpers.DefaultNamespace, fmt.Sprintf("service %s", serviceName)).Filter("{.spec.clusterIP}")
+			Expect(err).Should(BeNil())
+			Expect(govalidator.IsIP(res.String())).Should(BeTrue())
 
-	}, 300)
+			svcIP := res.String()
 
-	It("Check Service with cross-node", func() {
-		demoDSPath := kubectl.ManifestGet("demo_ds.yaml")
-		kubectl.Apply(demoDSPath)
-		defer kubectl.Delete(demoDSPath)
+			status := kubectl.Exec(fmt.Sprintf("curl http://%s/", svcIP))
+			status.ExpectSuccess()
 
-		waitPodsDs()
+			ciliumPod, err := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+			Expect(err).Should(BeNil())
 
-		svcIP, err := kubectl.Get(
-			helpers.DefaultNamespace, "service testds-service").Filter("{.spec.clusterIP}")
-		Expect(err).Should(BeNil())
-		log.Debugf("svcIP: %s", svcIP.String())
-		Expect(govalidator.IsIP(svcIP.String())).Should(BeTrue())
+			service := kubectl.CiliumExec(ciliumPod, "cilium service list")
+			Expect(service.Output()).Should(ContainSubstring(svcIP))
+			service.ExpectSuccess()
 
-		url := fmt.Sprintf("http://%s/", svcIP)
-		testHTTPRequest(url)
+		}, 300)
 	})
 
-	//TODO: Check service with IPV6
+	Context("Checks service across nodes", func() {
 
-	It("Check NodePort", func() {
-		demoDSPath := kubectl.ManifestGet("demo_ds.yaml")
-		kubectl.Apply(demoDSPath)
-		defer kubectl.Delete(demoDSPath)
+		var (
+			demoYAML = kubectl.ManifestGet("demo_ds.yaml")
+		)
 
-		waitPodsDs()
+		BeforeEach(func() {
```

Comment:
If you add in this context `BeforeAll` and `AfterAll` it'll be great

Reply:
Done, will merge once I get a green build!